code/script.js

```diff
@@ -231,95 +236,120 @@ const setSecret = () => {
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+        if (keep) {
+          alert(`Yes, the person does have ${value}. Keep all people with ${value}`);
+          charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+        } else {
+          alert(`No, the person does not have ${value}. Remove all people with ${value}`);
+          charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+        }
 
-  // Invoke a function to redraw the board with the remaining people.
+  } else if (category === 'hair') {
+        if (keep) {
+          alert(`Yes, the person does have ${value} hair. Keep all people with ${value} hair.`);
+          charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+        } else {
+          alert(`No, the person does not have ${value} hair. Remove all people with ${value} hair`);
+          charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+        }
+  } else if (category === 'eyes') {
+        if (keep) {
+          alert(`Yes, the person does have ${value} eyes. Keep all people with ${value} eyes`);
+          charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+        } else {
+          alert(`No, the person does not have ${value} eyes. Remove all people with ${value} eyes`);
+          charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+        }
+  } 
+  generateBoard(); // Redraw the game board with the filtered characters
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+// When clicking guess, the player first has to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  if (confirm(`Are you sure you want to guess ${personToConfirm}?`)) {
+    checkMyGuess(personToConfirm);
+  } else {
+    alert(`Okay, let's keep playing!`);
+  }
 }
 
-// If you confirm, this function is invoked
+// If player confirms, this function is invoked
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
-}
+  if (personToCheck === secret.name) {
+   winOrLoseText.innerHTML = `It was indeed ${personToCheck}! YOU WIN!`
+  } else {
+    winOrLoseText.innerHTML = `Nope, the right person was ${secret.name}. You lose 😞`
+  }
+  winOrLose.style.display = 'flex'
+  board.style.display = 'none'
```

Comment:
Nice to write it like this and put it here so it works for both the win and the lose scenario.